sklearn/impute.py

```diff
@@ -1265,3 +1271,281 @@ def fit_transform(self, X, y=None):
+        `missing_values` will be imputed.
+
+    n_neighbors : int, optional (default = 5)
+        Number of neighboring samples to use for imputation.
+
+    weights : str or callable, optional (default="uniform")
+        Weight function used in prediction.  Possible values:
+
+        - 'uniform' : uniform weights. All points in each neighborhood are
+          weighted equally.
+        - 'distance' : weight points by the inverse of their distance.
+          in this case, closer neighbors of a query point will have a
+          greater influence than neighbors which are further away.
+        - callable : a user-defined function which accepts an
+          array of distances, and returns an array of the same shape
+          containing the weights.
+
+    metric : str or callable, optional (default="nan_euclidean")
+        Distance metric for searching neighbors. Possible values:
+        - 'nan_euclidean'
+        - callable : a user-defined function which conforms to the
+          definition of _pairwise_callable(X, Y, metric, **kwds). The function
+          accepts two arrays, X and Y, and a ``missing_values`` keyword in
+          ``kwds`` and returns a scalar distance value.
+
+    row_max_missing : float, optional (default=0.5)
+        The maximum fraction of columns (i.e. features) that can be missing
+        before the sample is excluded from nearest neighbor imputation. It
+        means that such rows will not be considered a potential donor in
+        ``fit``, and in ``transform`` their missing feature values will be
+        imputed to be the column mean for the entire dataset.
+
+    col_max_missing : float, optional (default=0.8)
+        The maximum fraction of rows (or samples) that can be missing
+        for any feature beyond which an error is raised.
+
+    copy : boolean, optional(default=True)
+        If True, a copy of X will be created. If False, imputation will
+        be done in-place whenever possible. Note that, if metric is
+        "nan_euclidean" and copy=False then missing_values in the
+        input matrix X will be overwritten with zeros.
+
+    Attributes
+    ----------
+    statistics_ : array, shape (n_features, )
+        The 1-D array contains the mean of each feature calculated using
+        observed (i.e. non-missing) values. This is used for imputing
+        missing values in samples that are either excluded from nearest
+        neighbors search because they have too many ( > row_max_missing)
+        missing features or all of the sample's k-nearest neighbors
+        (i.e., the potential donors) also have the relevant feature value
+        missing.
+
+    References
+    ----------
+    * Olga Troyanskaya, Michael Cantor, Gavin Sherlock, Pat Brown, Trevor
+      Hastie, Robert Tibshirani, David Botstein and Russ B. Altman, Missing
+      value estimation methods for DNA microarrays, BIOINFORMATICS Vol. 17
+      no. 6, 2001 Pages 520-525.
+
```

Comment:
I think we wanted to note somewhere that a reference implementation in R behaves a bit differently, using less than K neighbours where some of the nearest neighbours have the target feature missing

Reply:
I am not familiar with the R implementation, is there a reference/link to it that discuss how it works?